### PR TITLE
Build: update esbuild to support cjs

### PR DESCRIPTION
**Pull request by WenheLI**
This PR is to address #520

### CODE REVIEW

**Comment by yorkie** [comment]
@FeelyChau @WenheLI Do you mind to add a workflow to confirm esbuild is working?

**Comment by WenheLI** [comment]
> @FeelyChau @WenheLI Do you mind to add a workflow to confirm esbuild is working?

I am working on the tests rn.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -20,14 +20,15 @@ on:
...
-    name: Build on node ${{ matrix.node_version }} and ${{ matrix.os }}
+    name: Build on node ${{ matrix.node_version }} and ${{ matrix.os }} with ${{ matrix.command }}
     if: ${{ github.event_name == 'push' || github.event.pull_request.draft == false }}
     runs-on: ${{ matrix.os }}
     timeout-minutes: 30
     strategy:
       matrix:
         node_version: ['12', '14']
         os: [ubuntu-latest, macOS-latest]
+        command: [build, dev-build]
```
Comment (yorkie) [inline .github/workflows/build.yml:31]: This added too many jobs, shall we just add a separate workflow to confirm node14+macOS? And not all steps in build.yml is needed, too :)

**Review by yorkie — CHANGES_REQUESTED** [review]

File: .github/workflows/dev-build.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
+name: dev-build
```
Comment (yorkie) [inline .github/workflows/dev-build.yml:1]: esbuild?

File: .github/workflows/dev-build.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node_version }}
+    - name: Building Pipcook
+      run: |
+        npm install
+        npm run build
+        npm run dev-build
+        npm run init-dev
+        du -h -d 1
```
Comment (yorkie) [inline .github/workflows/dev-build.yml:44]: We can stop from here :)

File: .github/workflows/dev-build.yml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    - '.github/markdown.yml'
+
+jobs:
+  universal:
+    name: Build on node ${{ matrix.node_version }} and ${{ matrix.os }}
+    if: ${{ github.event_name == 'push' || github.event.pull_request.draft == false }}
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 30
+    strategy:
+      matrix:
```
Comment (yorkie) [inline .github/workflows/dev-build.yml:28]: No necessary to use matrix.

File: .github/workflows/dev-build.yml [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    - master
+    paths-ignore:
+    - 'docker/**'
+    - 'docs/**'
+    - 'notebooks/**'
+    - '.github/markdown.yml'
+
+jobs:
+  universal:
+    name: Build on node ${{ matrix.node_version }} and ${{ matrix.os }}
```
Comment (yorkie) [inline .github/workflows/dev-build.yml:23]: Build with esbuild

File: .github/workflows/dev-build.yml [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    - uses: actions/checkout@v2
+  
+    - name: Using Node.js ${{ matrix.node_version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node_version }}
+    - name: Building Pipcook
+      run: |
+        npm install
+        npm run build
```
Comment (yorkie) [inline .github/workflows/dev-build.yml:41]: This should be removed.

File: .github/workflows/dev-build.yml [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+        node_version: ['14']
+        os: [ubuntu-latest]
+    steps:
+    - uses: actions/checkout@v2
+  
+    - name: Using Node.js ${{ matrix.node_version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node_version }}
+    - name: Building Pipcook
```
Comment (yorkie) [inline .github/workflows/dev-build.yml:38]: How about splitting this into multiple steps:

1. Install Pipcook
2. Build Pipcook
3. Test if the above build work

It displays the cost of every step and we could track it in every build.

**Review by yorkie — CHANGES_REQUESTED** [review]

File: .github/workflows/dev-build.yml (hunk 2 above)
Comment (yorkie) [reply .github/workflows/dev-build.yml:1]: dev-build just looks too general :)

File: .github/workflows/dev-build.yml (hunk 6 above)
Comment (WenheLI) [reply .github/workflows/dev-build.yml:41]: I think we should also run unit-test on the building output

File: .github/workflows/dev-build.yml (hunk 7 above)
Comment (WenheLI) [reply .github/workflows/dev-build.yml]: Done

File: .github/workflows/dev-build.yml (hunk 6 above)
Comment (yorkie) [reply .github/workflows/dev-build.yml]: The build won't run unit-tests IMO.

File: .github/workflows/dev-build.yml [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+jobs:
+  universal:
+    name: Build with esBuild
+    if: ${{ github.event_name == 'push' || github.event.pull_request.draft == false }}
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+    - uses: actions/checkout@v2
+
```
Comment (yorkie) [inline .github/workflows/dev-build.yml:29]: Unnecessary blank :)

File: .github/workflows/dev-build.yml [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    - name: Installing Pipcook
+      run: |
+        npm install
+    - name: Building Pipcook
+      run: |
+        npm run dev-build
+    - name: Running unit tests
+      run: |
+        npm run init-dev
+        npm test
```
Comment (yorkie) [inline .github/workflows/dev-build.yml:43]: Missing EOF

File: .github/workflows/dev-build.yml [hunk 10]
```diff
@@ -0,0 +1,42 @@
+name: esBuild
```
Comment (yorkie) [inline .github/workflows/dev-build.yml:1]: May I ask you to change the filename to `esbuild.yml` too?

**Review by yorkie — APPROVED** [review]
/cc @FeelyChau

**Pull request merged**